app/src/main/java/com/spentwell/viewmodels/SetEarningsViewModel.kt

```diff
@@ -1,21 +1,33 @@
 package com.spentwell.viewmodels
 
 import androidx.lifecycle.LiveData
+import androidx.lifecycle.MediatorLiveData
 import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
 import androidx.lifecycle.ViewModel
 
 class SetEarningsViewModel : ViewModel() {
 
+    val incomePerMonth = MutableLiveData<String>()
     private val _isProceedButtonEnabled = MutableLiveData<Boolean>()
     val isProceedButtonEnabled: LiveData<Boolean>
         get() = _isProceedButtonEnabled
 
+    private val observer = Observer { value: String ->
+        toggleProceedButton(value.isNotEmpty() && value.toInt() > 1000)
+    }
 
     init {
         _isProceedButtonEnabled.value = false
+        incomePerMonth.observeForever(observer)
     }
 
-    fun toggleProceedButton(b: Boolean) {
+    private fun toggleProceedButton(b: Boolean) {
```

Comment:
It would be helpful in future if you could rename the variable name from b to something else

Reply:
sure @shankar75031